runtime/lua/vim/lsp/extensions.lua

```diff
@@ -0,0 +1,48 @@
+local vim = vim
+local validate = vim.validate
+local log = require 'vim.lsp.log'
+local util = require 'vim.lsp.util'
+local buf = require 'vim.lsp.buf'
+
+local M = {}
+
+-- callback to preview a location in a floating window instead
+-- of jumping to it
+local function preview_location_callback(_, method, result)
+  if result == nil or vim.tbl_isempty(result) then
+    log.info(method, 'No location found')
+    return nil
+  end
+  if vim.tbl_islist(result) then
+    util.preview_location(result[1])
+  else
+    util.preview_location(result)
+  end
+end
+
+local function peek_request(method, params)
+  validate {
+    method = {method, 's'};
+  }
+  return vim.lsp.buf_request(0, method, params, preview_location_callback)
+end
+
+function M.peek_definition()
+  local params = util.make_position_params()
+  peek_request('textDocument/definition', params)
```

Comment:
can you return the result of peek request (just so it's consistent with other requests) and also no need to make a temp params, just pass params in to the peek request.

Reply:
Well, I tried to be consistent with the other requests, but then you started asking for changes ;) (compare `M.definition()` in `buf.lua`.)